src/durationHelper.ts

```diff
@@ -0,0 +1,185 @@
+/*
+ *  Power BI Visualizations
+ *
+ *  Copyright (c) Microsoft Corporation
+ *  All rights reserved.
+ *  MIT License
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the ""Software""), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+
+module powerbi.extensibility.visual {
+    const GanttDurationUnitType = [
+        "second",
+        "minute",
+        "hour",
+        "day",
+    ];
+
+    const HoursInADay: number = 24;
+    const MinutesInAHour: number = 60;
+    const SecondsInAMinute: number = 60;
+    const MinutesInADay: number = 24 * MinutesInAHour;
+    const SecondsInADay: number = 60 * MinutesInADay;
+    const SecondsInAHour: number = MinutesInAHour * SecondsInAMinute;
+
+    export class DurationHelper {
+
+        public static getNewUnitByFloorDurationFloor(durationUnitTypeIndex: number, duration: number): string {
```

Comment:
please rename this method, **floor** is present twice in getNewUnitByFloorDurationFloor